### ISSUE TITLE
About the training method for evaluation Oracle and the details of dividing the training in-distribution set and  valid out-of-distribution set.

### ISSUE THREAD

**Issue by Mistletoe-git** [ChenyuWang-Monica/DRAKES#2]
Hello, first of all, thank you for your excellent work!

Currently, I am trying to reproduce the content about protein inverse folding in your paper and have made good progress.

However, regarding the fine-tuning Oracle and evaluation Oracle mentioned in the paper, I still have some confusion, please forgive my ignorance.

From the code `train_oracle.py`, the difference between the two is only in the content included in the “training set”. For example, the evaluation Oracle includes all content from the training set, validation set, and test set.

In the paper, you mentioned using the in-distribution set and out-of-distribution set for training and validation, respectively. Regarding this content, it seems that it is not reflected in the code. Or perhaps you have already completed the dataset preprocessing beforehand, so I would like to know more details about this issue. How are the I/O sets specifically processed?

Looking forward to your reply!

**Comment by ChenyuWang-Monica** [comment]
Thank you for your interest!

The final fine-tuning oracle and evaluation oracle used in fine-tuning and evaluation in the paper does not involve additional splitting, so you can get them directly via `train_oracle.py`.

The I/O sets mentioned in the appendix is an additional check for the performance of these oracles, which uses random splitting of the data to 80%/20%. This is only for testing the performance in Table 8 and select training hyperparameters (eg. number of epochs), and is not used as the final fine-tuning oracle and evaluation oracle used in the main paper.
